File: js-solutions/src/day2/part2.js
import { stringInputToArray } from '../utils/string-input-to-array.js';
import { data } from './input2.js';

const dataArray = stringInputToArray(data);
const answer = dataArray.filter((password) => {
    const [lengthReq, letter, pwd] = password.split(' ');
    let [pwdMin, pwdMax] = lengthReq.split('-')
    pwdMin--;
    pwdMax--;
    const letterReq = letter.replace(':', '')

    let isPwdValid = false

    if (pwdMin < pwd.length && pwdMin >= 0) {
        if (pwd[pwdMin] == letterReq) {
            isPwdValid = true
        }
    }
    if (pwdMax < pwd.length && pwdMax >= 0) {
        if (pwd[pwdMax] === letterReq) {
            if (isPwdValid) {
                isPwdValid = false
            }
            else {
                isPwdValid = true
            }
        }
    }

    return isPwdValid;
});

console.log(answer.length)

File: day2/part2.py
file = open("input2.txt", 'r')
lines = file.readlines()
countPwdRespectRule = 0

for l in lines:
    words = l.split(' ')
    pwdLengthRule = words[0]
    pwdMustContainLetter = words[1].replace(':', '')
    pwd = words[2].replace('\n', '')
    # print(pwdLengthRule, pwdLetterRule, pwd)
    pwdLengthRule = pwdLengthRule.split('-')
    pwdMin = int(pwdLengthRule[0]) - 1
    pwdMax = int(pwdLengthRule[1]) - 1
    
    isPwdValid = False
    if pwdMin < len(pwd) and pwdMin >= 0:
        if pwd[pwdMin] == pwdMustContainLetter:
            isPwdValid = True
    
    if pwdMax < len(pwd) and pwdMax >= 0:
        if pwd[pwdMax] == pwdMustContainLetter:
            if isPwdValid:
                isPwdValid = False
            else:
                isPwdValid = True


    if isPwdValid:
        countPwdRespectRule = countPwdRespectRule + 1

print(countPwdRespectRule)